spinn_front_end_common/utilities/helpful_functions.py

```diff
@@ -16,6 +16,6 @@ def read_and_convert(x, y, address, length, data_format, transceiver):
     """
 
     # turn byte array into str for unpack to work
-    data = str(list(transceiver.read_memory(x, y, address, length))[0])
-    result = struct.unpack(data_format, data)[0]
+    data = transceiver.read_memory(x, y, address, length)
+    result = struct.unpack_from(data_format, data)
```

Comment:
we should keep the [0] from the result line. the only calls to this end up doing it there, which makes it a bit ugly. Having it here simplies the code all around.